Mathlib/RingTheory/MvPolynomial/NewtonIdentities.lean

```diff
@@ -0,0 +1,303 @@
+/-
+Copyright (c) 2023 Michael Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Lee
+-/
+import Mathlib.Algebra.Algebra.Subalgebra.Basic
+import Mathlib.Data.Finset.Card
+import Mathlib.Data.Fintype.Basic
+import Mathlib.Data.MvPolynomial.CommRing
+import Mathlib.Data.MvPolynomial.Rename
+import Mathlib.Data.Nat.Parity
+import Mathlib.RingTheory.MvPolynomial.Basic
+import Mathlib.RingTheory.MvPolynomial.Symmetric
+import Mathlib.RingTheory.Polynomial.Basic
+
+/-!
+# Newton's Identities
+
+This file defines `MvPolynomial` power sums as a means of implementing Newton's identities.
+
+## Main declarations
+
+* `MvPolynomial.psum`: The degree-`n` power sum evaluated at the elements of `s`.
+
+* `MvPolynomial.NewtonIdentities.esymm_recurrence`: a recurrence relation for the kth elementary
+  symmetric polynomial in terms of lower degree elementary symmetric polynomials and power sums.
+
+## Notation
+
++ `psum σ R n` is the degree-`n` power sum in `MvPolynomial σ R`, i.e. the sum of monomials
+  `(X i)^n` over `i ∈ σ`.
+
+## References
+
+See [zeilberger1984] for the combinatorial proof of Newton's identities.
+-/
+
+open Equiv (Perm)
+
+open BigOperators MvPolynomial
+
+noncomputable section
+
+namespace Multiset
+
+variable {R : Type _} [CommSemiring R]
+
+/-- The degree-`n` power sum evaluated at the elements of `s` -/
+def psum (s : Multiset R) (n : ℕ) : R := (s.map (· ^ n : R → R)).sum
+
+end Multiset
+
+namespace MvPolynomial
+
+section PowerSum
+
+open Finset
+
+variable (σ R) [CommSemiring R] [Fintype σ] [Fintype τ]
+
+/-- The degree-`n` power sum -/
+def psum (n : ℕ) : MvPolynomial σ R := ∑ i, X i ^ n
+
+@[simp]
+theorem psum_zero : psum σ R 0 = card (@univ σ _) := by
```

Comment:
this is usually spelt as `Fintype.card \sigma`! (bigger review later)

Also, I'd say probably preferrable to use `univ (\alpha = \sigma)` instead of `@`, for other contexts' reference

Reply:
Fixed!